mijia-homie/src/main.rs

```diff
@@ -289,19 +295,20 @@ async fn bluetooth_connection_loop(
 
         // Check the state of each sensor and act on it if appropriate.
         {
-            let ids: Vec<DeviceId> = state.lock().await.sensors.keys().cloned().collect();
-            for id in ids {
+            let mac_addresses: Vec<MacAddress> =
+                state.lock().await.sensors.keys().cloned().collect();
+            for mac_address in mac_addresses {
                 let connection_status = state
                     .lock()
                     .await
                     .sensors
-                    .get(&id)
+                    .get(&mac_address)
                     .map(|sensor| {
                         log::trace!("State of {} is {:?}", sensor.name, sensor.connection_status);
-                        sensor.connection_status
+                        sensor.connection_status.to_owned()
```

Comment:
is the adapter id (that's in the DeviceId string) always an integer? If so, maybe we *could* make DeviceId Copy, and convert it to a string only when we need to?

Reply:
It's a string. In the cases I've seen so far it's something like `hci0`. How would you store `DeviceId` instead?